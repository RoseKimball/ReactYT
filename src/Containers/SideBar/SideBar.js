import React, { Component } from 'react';
import { Menu, Divider } from 'semantic-ui-react';
import './SideBar.scss';
import SideBarItem from '../../Components/SideBarItem/SideBarItem';
import SideBarHeader from '../../Components/SideBarHeader/SideBarHeader';
import Subscriptions from './Subscriptions/Subscriptions';
import SideBarFooter from '../../Components/SideBarFooter/SideBarFooter';

export class SideBar extends Component {
    render() {
        return(
            <Menu borderless vertical stackable fixed='left' className='side-nav'>
                    <SideBarItem highlight={true} className='highlight-item'  label='Home' icon='home'/>
                    <SideBarItem label='Trending' icon='fire'/>
                    <SideBarItem label='Followers' icon='spy'/>
                    <Divider />
                    <SideBarHeader title='Library'/>
                    <SideBarItem label='History' icon='history'/>
                    <SideBarItem label='Watch later' icon='clock'/>
                    <SideBarItem label='Liked videos' icon='thumbs up'/>
                    <Divider />
                    <Subscriptions />
                    <Divider />
                    <SideBarHeader title='More From Youtube'/>
                    <SideBarItem label='Movies and Shows' icon='film'/>
                    <Divider />
                    <SideBarItem label='Report history' icon='flag'/>
                    <SideBarItem label='Help' icon='help circle'/>
                    <SideBarItem label='Send feedback' icon='comment'/>
                    <Divider />
                    <SideBarFooter />
            </Menu>
        );
    };
};

export default SideBar;