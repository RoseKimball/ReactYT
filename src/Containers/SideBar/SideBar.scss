@import '../../styles/Shared.scss';

.ui.menu.fixed.side-nav {
    background-color: #f5f5f5;
    margin-top: $header-height;
    overflow-y: auto;
    padding-bottom: $header-height;

    .sidebar-item:hover {
        background: #ebebeb;
    }
}

.side-nav.ui.vertical.menu {
   width: $sidebar-width;
}