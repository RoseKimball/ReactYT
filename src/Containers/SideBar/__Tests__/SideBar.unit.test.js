import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import SideBar from '../SideBar';

describe('SideBar', () => {
    test('renders SideBar', () => {
        const wrapper = shallow(
            <SideBar />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

