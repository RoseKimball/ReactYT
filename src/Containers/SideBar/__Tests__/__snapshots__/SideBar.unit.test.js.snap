// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders SideBar 1`] = `ShallowWrapper {}`;
